### ISSUE TITLE
Question about Get requests

### ISSUE THREAD

**Issue by nidhaloff** [libcpr/cpr#669]
I'm searching for a lib to create an http client in c++. This lib looks perfect, however, for some reason, I'm not getting a response when making a very simple get request using a fake JSON generator. This is just a question from my side, I don't believe it's a bug in the lib so I'm just asking here.

## Description

Here is my very simple example:

```
#include <cpr/cpr.h>
#include <iostream>

int main(int argc, char** argv) {
cpr::Response r = cpr::Get(cpr::Url{"https://jsonplaceholder.typicode.com/posts"});
std::cout << r.status_code << std::endl;  // should print something but it doesn't
std::cout << r.url << std::endl;  // should print something but it doesn't
std::cout << r.text << std::endl;  // should print something but it doesn't

}
```

## Expected Behavior
Well I expected to receive a response, meaning r.url, r.text should have a value

## Actual Behavior
I receive nothing back, meaning r.url and r.text are not printed/they have no value.

## Your Environment
* Version used:1.6.2
* Where did you get it from (e.g. conan, vcpkg, master, ...): used the FetchContent example in the readme
* Operating System and version: ubuntu 20.04

**Comment by KingKili** [comment]
Thanks for the good first issue. It seems like something with OpenSSL is broken again. As a quick workaround until this issue is resolved I'd recommend using the ubuntu curl version from apt (`libcurl-dev`). You than can force CMake to use os curl during configuration phase using  `cmake .. -DCPR_FORCE_USE_SYSTEM_CURL=ON`

**Comment by KingKili** [comment]
I have created a PR which fixes the issue for me. Once it is merged, please let me know if it worked for you too.

**Comment by nidhaloff** [comment]
@KingKili wow thanks, you are right.

Steps that I made for this to work:
- sudo apt install libcurl4-openssl-dev
- cmake .. -DCPR_FORCE_USE_SYSTEM_CURL=ON

**Comment by nidhaloff** [comment]
@KingKili I'm working on the same example but now I'm using conan to build the project instead of "FetchContent" (readme example), but I get segmentation fault when I run the binary. Do you maybe have any ideas why is this happening?

**Comment by KingKili** [comment]
@nidhaloff Sounds like #634. The fix was adding `conan_basic_setup()` to the `CMakeLists`. Does it help?

**Comment by nidhaloff** [comment]
@KingKili thanks for the suggestion, but I already done that. Here is my cmake setup:


```
cmake_minimum_required (VERSION 3.10)
project(hello CXX)

add_executable(${PROJECT_NAME} src/main.cpp)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake) # Include Conan-generated file
conan_basic_setup(TARGETS) # Introduce Conan-generated targets

target_link_libraries(${PROJECT_NAME} CONAN_PKG::cpr)

```

If I remove (TARGETS) from conan_basic_setup then the build fails:
```
CMake Error at CMakeLists.txt:5 (add_executable):
  Target "hello" links to target "CONAN_PKG::cpr" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


-- Generating done
```

**Comment by KingKili** [comment]
Strange than try adding `add_definitions(-D_GLIBCXX_USE_CXX11_ABI=0)
`

**Comment by nidhaloff** [comment]
@KingKili Thanks, It worked! 
I'm not so familiar with cmake, can you please explain what does that line does or rather what the problem was?

**Comment by KingKili** [comment]
Essentially conan and CMake use different libcxx ABI versions. If your are more interrested feel free to take look at #634 especially [here](https://github.com/libcpr/cpr/issues/634#issuecomment-939153553).
